From 20e29561a9ea0d67de5fd42732dac95527cb2264 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Fri, 27 Jan 2023 14:19:23 +0100
Subject: [PATCH 090/954] refactor(st-bsec): remove bsec_write_debug_conf

Remove the unused function bsec_write_debug_conf, the debug configuration
should be not modified by TF-A.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 130d9ce83ab91de192b08605a6a28b83edcb98e0 refactor(st-bsec): remove bsec_write_debug_conf]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I728a175872c8e6f07025b2212ffb54e27cbe43ae
---
 drivers/st/bsec/bsec2.c   | 19 +------------------
 include/drivers/st/bsec.h |  3 +--
 2 files changed, 2 insertions(+), 20 deletions(-)

diff --git a/drivers/st/bsec/bsec2.c b/drivers/st/bsec/bsec2.c
index ee5e780a48..ca7956d142 100644
--- a/drivers/st/bsec/bsec2.c
+++ b/drivers/st/bsec/bsec2.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2017-2022, STMicroelectronics - All Rights Reserved
+ * Copyright (c) 2017-2023, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -580,23 +580,6 @@ uint32_t bsec_permanent_lock_otp(uint32_t otp)
 	return result;
 }
 
-/*
- * bsec_write_debug_conf: write value in debug feature.
- *	to enable/disable debug service.
- * val: value to write.
- * return value: none.
- */
-void bsec_write_debug_conf(uint32_t val)
-{
-	if (is_otp_invalid_mode()) {
-		return;
-	}
-
-	bsec_lock();
-	mmio_write_32(bsec_base + BSEC_DEN_OFF, val);
-	bsec_unlock();
-}
-
 /*
  * bsec_read_debug_conf: return debug configuration register value.
  */
diff --git a/include/drivers/st/bsec.h b/include/drivers/st/bsec.h
index 60dcf3c12f..a96570cf8a 100644
--- a/include/drivers/st/bsec.h
+++ b/include/drivers/st/bsec.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2017-2022, STMicroelectronics - All Rights Reserved
+ * Copyright (c) 2017-2023, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -102,7 +102,6 @@ uint32_t bsec_write_otp(uint32_t val, uint32_t otp);
 uint32_t bsec_program_otp(uint32_t val, uint32_t otp);
 uint32_t bsec_permanent_lock_otp(uint32_t otp);
 
-void bsec_write_debug_conf(uint32_t val);
 uint32_t bsec_read_debug_conf(void);
 
 void bsec_write_scratch(uint32_t val);
-- 
2.53.0

